src/rate-limiting/simple-counting.ts

```diff
@@ -65,24 +60,51 @@ export class SimpleCountingRateLimiter implements RateLimiter {
       this.requestsThisMinute = 0
     }
 
+    return {
+      now,
+      nextSecondInterval,
+      nextMinuteInterval,
+    }
+  }
+
+  msUntilNextExecution(): number {
+    // If the limit is set to infinity, there was no tier limit specified
+    if (this.perSecondLimit === Infinity && this.perMinuteLimit === Infinity) {
+      return 0
+    }
+
+    const { now, nextSecondInterval, nextMinuteInterval } = this.updateIntervals()
```

Comment:
I couldn't put the comment further up but weren't we going to consolidate the simple counting rate limits to just seconds? Because of scenarios like `RATE_LIMIT_CAPACITY_SECOND: 10` and `RATE_LIMIT_CAPACITY_MINUTE: 50`. We'd blow the credits within the first 5 seconds and have to wait till the next minute causing data to be stale over the next 55s.

Reply:
That was an alternative to the fixed rate limiting, with the different statuses it's ok if that's what happens with the counting strategy. In reality, this strategy should be used when the rate limiting on the provider comfortably allows for the regular load on the adapter, and the bursts would cover a random flurry of messages. I do agree that there might be "smarter" ways of handling it, but those should be implemented as separate strategies imo

Reply:
If the adapter config allows regular load and bursts, I think consolidating to seconds would still allow for bursts to be used effectively. It'd just cover the scenarios that the longer term tier is lacking a little so we'd need smaller bursts over time. Incase someone accidentally configures `counting` without realizing how it works under the hood, consolidating feels safer. But I understand if we want to keep the large bursts up front so I'll leave it up to you if you want to resolve this comment.